Providers/nxOMSAutomationWorker/automationworker/3.x/worker/urllib2httpclient.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python
+# ====================================
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+# ====================================
+"""Urllib2 HttpClient."""
+
+import httplib
+import socket
+import time
+import traceback
+import urllib2
+
+from httpclient import *
+from workerexception import *
+
+PY_MAJOR_VERSION = 0
+PY_MINOR_VERSION = 1
+PY_MICRO_VERSION = 2
+
+SSL_MODULE_NAME = "ssl"
+
+# On some system the ssl module might be missing
+try:
+    import ssl
+except ImportError:
+    ssl = None
+
+
+class HttpsClientHandler(urllib2.HTTPSHandler):
+    """Https handler to enable attaching cert/key to request. Also used to disable strict cert verification for
+    testing.
+    """
+
+    def __init__(self, cert_path, key_path, insecure=False):
+        self.cert_path = cert_path
+        self.key_path = key_path
+
+        ssl_context = None
+        if insecure and SSL_MODULE_NAME in sys.modules and (sys.version_info[PY_MAJOR_VERSION] == 2 and
```

Comment:
is this required ?

Reply:
Yes, when python2 runbooks are executed in python3 worker environment

Reply:
This scenario is not possible today. Need to find a way to test this scenario locally to be sure all code works fine and we can make changes smoothly in future.